### PR TITLE
Improvement of Doxycomments in the CCSDS FDP checksum class

### PR DESCRIPTION
**Pull request by JackNWhite**
| | |
|:---|:---|
|**_Related Issue(s)_**| n |
|**_Has Unit Tests (y/n)_**| n |
|**_Documentation Included (y/n)_**| y |

---
## Change Description

- General improvement of the class Doxycomments, in particular those of the class itself and of the `update()` method.
- Fixed/reinstated a broken link in the contributions readme.

## Rationale

Usage of the class, particularly the `update()` method, was ambiguous and the existing comments were unclear.

### CODE REVIEW

**Review by LeStarch — CHANGES_REQUESTED** [review]
I added a comment about consistency for block comments.

File: CFDP/Checksum/Checksum.hpp [hunk 1]
```diff
@@ -17,9 +17,40 @@
 
 namespace CFDP {
 
-  //! \class Checksum
-  //! \brief Class representing a CFDP checksum
-  //!
+  /*! \class Checksum
```
Comment (LeStarch) [inline CFDP/Checksum/Checksum.hpp]: I see two different styles in this file for block comments:

```
/*!
 *
 */
```

And:

```
//!
//!
//!
```
Can we converge on one?  F Prime is converging on the second one (`//!`)

Other than that, this looks good!

**Review by LeStarch — CHANGES_REQUESTED** [review]

**Review by thomas-bc — APPROVED** [review]
I just recently watched a talk about the power and importance of documentation so thank you very much for this!

**Comment by LeStarch** [comment]
CI failure is being investigated.

File: CFDP/Checksum/Checksum.hpp (hunk 1 above)
Comment (JackNWhite) [reply CFDP/Checksum/Checksum.hpp]: I switched the class comment to line comment style.

**Review by thomas-bc — APPROVED** [review]

**Review by LeStarch — APPROVED** [review]

**Comment by LeStarch** [comment]
Ignoring known CI failure.

**Pull request merged**